config/master.d/50-master.conf

```diff
@@ -22,6 +22,11 @@ pillar_roots:
   base:
     - /usr/share/salt/kubernetes/pillar
 
+# force the modules path in the master so not even
+# a "sync_modules" is necessary
+module_dirs:
+  - /usr/share/salt/kubernetes/salt/_modules
```

Comment:
I think this can be preventing from `_pillar` to be autodetected and probably other modules.

Reply:
I don't think that could be the case. The pillar seems to be loading fine.

Reply:
You are right, I mistook it for `extension_modules`.

Running this PR locally to check if it works here.